File: data/readable_format/cpp/file1613.cpp
#include <iostream>
#include <cmath> // For std::abs

// Declaration of the QuadraticParameters class
class QuadraticParameters {
public:
    double a, b, c; // Parameters a, b, c

    // Constructor to initialize parameters with default values
    QuadraticParameters() : a(0.0), b(0.0), c(0.0) {}

    // Method to set parameters a, b, c
    void set(double new_a, double new_b, double new_c) {
        a = new_a;
        b = new_b;
        c = new_c;
    }

    // Method to calculate the quadratic value
    double quadratic(double x) {
        return a * x * x + b * x + c;
    }
};

int main() {
    // Create an instance of the QuadraticParameters class
    QuadraticParameters qp;
    
    // Set the parameters: a=1, b=-3, c=2 (roots are x=1 and x=2)
    qp.set(1.0, -3.0, 2.0);
    
    // Test value
    double x = 1.0;
    
    // Calculate the result
    double result = qp.quadratic(x);
    
    // Check the result (allowing for a small numerical error margin)
    const double epsilon = 1e-7;
    if (std::abs(result - 0.0) < epsilon) {
        std::cout << "Test passed: quadratic(1) = 0" << std::endl;
    } else {
        std::cout << "Test failed: quadratic(1) != 0, result = " << result << std::endl;
    }

    return 0;
}

File: data/readable_format/fortran/file1613.f90
program main
    implicit none
    ! Test variables
    double precision :: x, result

    ! Initialize parameters
    call quadratic_set(1.0d0, -3.0d0, 2.0d0)  ! a=1, b=-3, c=2, roots are x=1 and x=2
    x = 1.0d0

    ! Calculate result
    result = quadratic(x)

    ! Check the result
    if (result == 0.0d0) then
        print *, "Test passed: quadratic(1) = 0"
    else
        print *, "Test failed: quadratic(1) != 0, result = ", result
    end if

contains

    subroutine quadratic_set(new_a, new_b, new_c)
        implicit none
        double precision, intent(in) :: new_a, new_b, new_c
        double precision :: a, b, c

        a = new_a
        b = new_b
        c = new_c
    end subroutine quadratic_set

    function quadratic(x) result(quadratic_result)
        implicit none
        double precision, intent(in) :: x
        double precision :: quadratic_result
        double precision :: a, b, c

        quadratic_result = a*x**2 + b*x + c
    end function quadratic

end program main